core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```
@@ -0,0 +1,659 @@
+   */
+  protected ObjectUnderFileSystem(AlluxioURI uri) {
+    super(uri);
+  }
+
+  /**
+   * Information about a single object in object UFS.
+   */
+  protected class ObjectStatus {
+    long mFileSize;
+    long mLastModifiedTimeMs;
+
+    public ObjectStatus(long fileSize, long lastModifiedTimeMs) {
+      mFileSize = fileSize;
+      mLastModifiedTimeMs = lastModifiedTimeMs;
+    }
+
+    /**
+     * Gets the file size in bytes.
+     *
+     * @return the file size in bytes
+     */
+    public long getContentLength() {
+      return mFileSize;
+    }
+
+    /**
+     * Gets the UTC last modified time in ms.
+     *
+     * @return modification time in milliseconds
+     */
+    public long getLastModifiedTimeMs() {
+      return mLastModifiedTimeMs;
+    }
+  }
+
+  /**
+   * A chunk of listing results.
+   */
+  public interface ObjectListingResult {
+    /**
+     * Objects in a pseudo-directory which may be a file or a directory.
+     *
+     * @return a list of objects
+     */
+    String[] getObjectNames();
+
+    /**
+     * Use common prefixes to infer pseudo-directories in object store.
+     *
+     * @return a list of common prefixes
+     */
+    String[] getCommonPrefixes();
+
+    /**
+     * Get next chunk of object listings.
+     *
+     * @return null if listing did not find anything or is done, otherwise return new
+     * {@link ObjectListingResult} for the next chunk
+     * @throws IOException if a non-alluxio error occurs
+     */
+    ObjectListingResult getNextChunk() throws IOException;
+  }
+
+  @Override
+  public void close() throws IOException {
+  }
+
+  @Override
+  public void connectFromMaster(String hostname) {
+  }
+
+  @Override
+  public void connectFromWorker(String hostname) {
+  }
+
+  @Override
+  public OutputStream create(String path, CreateOptions options) throws IOException {
+    if (mkdirs(getParentPath(path))) {
+      return createObject(stripPrefixIfPresent(path));
+    }
+    return null;
+  }
```

Should an exception be thrown instead of returning null?